### PR TITLE
applications: asset_tracker_v2: enable native_posix build

### PR DESCRIPTION
**Pull request by maxd-nordic**
This patch enables the Asset Tracker to be built for native_posix for test purposes.

Signed-off-by: Jørgen Kvalvaag <jorgen.kvalvaag@nordicsemi.no>
Signed-off-by: Maximilian Deubel <maximilian.deubel@nordicsemi.no>

### CODE REVIEW

**Review by jtguggedal — CHANGES_REQUESTED** [review]
I think we should add `native_posix` as integration platform and allowed platform in `sample.yaml`

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c [hunk 1]
```diff
@@ -19,6 +19,10 @@
 #include <modem/modem_info.h>
 #include "cJSON_os.h"
 
+#ifndef EFTYPE
+#define EFTYPE EINVAL
```
Comment (jayteemo) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c:23]:
```suggestion
#define EFTYPE EPROTO
```
is EPROTO available?  
EINVAL is already used as an error return value in the same functions as EFTYPE.

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c [hunk 2]
```diff
@@ -16,6 +16,10 @@
 #include <net/nrf_cloud_pgps.h>
 #endif
 
+#ifndef EFTYPE
```
Comment (jayteemo) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c:19]: can you instead change the return value of `static int location_process(const char * const buf)` from EFTYPE to ENOSYS?

**Review by plskeggs — APPROVED** [review]

**Review by jorgenmk — CHANGES_REQUESTED** [review]
In `src/events/debug_module_event.h` these event type names should be renamed:
`DEBUG_EVT_QEMU_X86_INITIALIZED`
`DEBUG_EVT_QEMU_X86_NETWORK_CONNECTED`
To something like `DEBUG_EVT_EMULATOR_<...>`

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c (hunk 1 above)
Comment (maxd-nordic) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c]: I just replaced the occurrences with EPROTO. :+1:

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c (hunk 2 above)
Comment (maxd-nordic) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c]: done

**Comment by maxd-nordic** [comment]
> In `src/events/debug_module_event.h` these event type names should be renamed: `DEBUG_EVT_QEMU_X86_INITIALIZED` `DEBUG_EVT_QEMU_X86_NETWORK_CONNECTED` To something like `DEBUG_EVT_EMULATOR_<...>`

done

**Review by jorgenmk — APPROVED** [review]

**Review by jayteemo — APPROVED** [review]

**Review by glarsennordic — APPROVED** [review]

**Review by jtguggedal — APPROVED** [review]

File: applications/asset_tracker_v2/prj_native_posix.conf [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+CONFIG_CLOUD_CLIENT_ID_USE_CUSTOM=y
+CONFIG_CLOUD_CLIENT_ID="cloud-id"
+
+# Disable unsupported application modules
+CONFIG_LOCATION_MODULE=n
+CONFIG_SENSOR_MODULE=n
+CONFIG_MODEM_MODULE=n
+CONFIG_UI_MODULE=n
+
+# Running the application in QEMU x86 emulator requires functionality
```
Comment (simensrostad) [inline applications/asset_tracker_v2/prj_native_posix.conf:57]:
```suggestion
# Running the application in Native Posix emulator requires functionality
```

File: applications/asset_tracker_v2/src/events/debug_module_event.h [hunk 4]
```diff
@@ -34,12 +34,12 @@ enum debug_module_event_type {
 	 *  MODEM_EVT_INITIALIZED which is not sent due to the modem module being disabled for
 	 *  QEMU x86 builds.
 	 */
-	DEBUG_EVT_QEMU_X86_INITIALIZED,
```
Comment (simensrostad) [inline applications/asset_tracker_v2/src/events/debug_module_event.h:37]: The event description needs an update to reflect the event renaming.

File: applications/asset_tracker_v2/src/events/debug_module_event.h [hunk 5]
```diff
@@ -34,12 +34,12 @@ enum debug_module_event_type {
...
+	DEBUG_EVT_EMULATOR_INITIALIZED,
 
 	/** Event sent when the application is built for QEMU x86.
 	 *  When built for QEMU x86 it is assumed that the application is connected to the network.
 	 */
-	DEBUG_EVT_QEMU_X86_NETWORK_CONNECTED,
+	DEBUG_EVT_EMULATOR_NETWORK_CONNECTED,
```
Comment (simensrostad) [inline applications/asset_tracker_v2/src/events/debug_module_event.h:42]: Here also.

File: applications/asset_tracker_v2/prj_native_posix.conf (hunk 3 above)
Comment (maxd-nordic) [reply applications/asset_tracker_v2/prj_native_posix.conf]: done

File: applications/asset_tracker_v2/src/events/debug_module_event.h (hunk 4 above)
Comment (maxd-nordic) [reply applications/asset_tracker_v2/src/events/debug_module_event.h:37]: done

File: applications/asset_tracker_v2/tests/debug_module/src/debug_module_test.c [hunk 6]
```diff
@@ -70,13 +70,15 @@ void setUp(void)
 	cmock_watchdog_app_Init();
 	cmock_modules_common_Init();
 	cmock_app_event_manager_Init();
+	cmock_app_event_manager_priv_Init();
 }
```
Comment (simensrostad) [inline applications/asset_tracker_v2/tests/debug_module/src/debug_module_test.c:74]: Apparently init and verify of mocks happens automatically. So the `setUp` and `tearDown` functions can be removed.
cc @balaji-nordic

**Review by simensrostad — APPROVED** [review]

File: applications/asset_tracker_v2/tests/debug_module/src/debug_module_test.c (hunk 6 above)
Comment (maxd-nordic) [reply applications/asset_tracker_v2/tests/debug_module/src/debug_module_test.c:74]: There will be a separate PR to remove those.

**Review by divipillai — APPROVED** [review]

**Pull request merged**